File: src/main/kotlin/br/com/zup/beagleui/framework/ui/DropDown.kt
package br.com.zup.beagleui.framework.ui

import br.com.zup.beagleui.framework.core.DumbWidget
import br.com.zup.beagleui.framework.core.Widget

data class DropDown(
    val header: Widget,
    val child: Widget
) : DumbWidget()

File: Sources/BeagleUI/Sources/DeclarativeViews/UIComponents/DropDown.swift
//
//  DropDown.swift
//  BeagleUI
//
//  Created by Daniel Tes on 12/09/19.
//  Copyright © 2019 Daniel Tes. All rights reserved.
//

public struct DropDown: UIComponentWidget {
    
    // MARK: - Public Properties
    
    public let header: Widget
    public let child: Widget
    
    // MARK: - Initialization
    
    init(
        header: Widget,
        child: Widget
    ) {
        self.header = header
        self.child = child
    }
    
    public init(
        header headerBuilder: () -> Widget,
        child childBuilder: () -> Widget
    ) {
        header = headerBuilder()
        child = childBuilder()
    }
    
}
